src/nlsolve/broyden.jl

```diff
@@ -22,7 +22,7 @@ end
 
 __get_linesearch(::SimpleBroyden{LS}) where {LS} = Val(LS)
 
-function SciMLBase.__solve(prob::NonlinearProblem, alg::SimpleBroyden, args...;
+function SciMLBase.__solve(prob::Union{NonlinearProblem, ImmutableNonlinearProblem}, alg::SimpleBroyden, args...;
```

Comment:
None of these unions should be required. If done correctly, the top level should change to an ImmutableNonlinearProblem, and then they should all be that.